Runtime/Scripts/MobiledgeXUDPClient.cs

```diff
@@ -0,0 +1,144 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+using System.Collections.Concurrent;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using System.Linq;
+namespace MobiledgeX
+{
+    // MobiledgeXUDPClient is a UDP Client Implementation offered with MobiledgeX Unity Package
+    // MobiledgeXUDPClient concurrency model supports the use of a single queue for
+    // send, and another queue for recieve. MobiledgeXUDPClient here has 1 independent thread
+    // per send or receive direction of communication.
+    public class MobiledgeXUDPClient : IDisposable
+    {
+        private UdpClient udpClient;
+        private string host;
+        private int port;
+        Thread receiveThread { get; set; }
+        Thread sendThread { get; set; }
+        static UTF8Encoding encoder;
+        public ConcurrentQueue<byte[]> receiveQueue { get; }
+        public BlockingCollection<ArraySegment<byte>> sendQueue { get; }
+        public bool run = true;
+        const int MAXPAYLOADSIZE = 508; // max payload size guaranteed to be deliverable (not guaranteed to be delivered)
+
+        public MobiledgeXUDPClient(string hostName, int sendPort, int receivePort)
+        {
+            try
+            {
+                udpClient = new UdpClient(receivePort);
+                udpClient.DontFragment = true;
+
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Failed to listen to UDP Messages at port " + receivePort + ": " + e.Message);
+                return;
+            }
+
+            host = Dns.GetHostAddresses(hostName)
+                .First(ip => ip.AddressFamily == AddressFamily.InterNetwork
+                || ip.AddressFamily == AddressFamily.InterNetworkV6).ToString();
+            port = sendPort;
+            encoder = new UTF8Encoding();
+            receiveQueue = new ConcurrentQueue<byte[]>();
+            receiveThread = new Thread(RunReceive);
+            receiveThread.Start();
+            sendQueue = new BlockingCollection<ArraySegment<byte>>();
+            sendThread = new Thread(RunSend);
+            sendThread.Start();
+            Connect();
+        }
+
+        public void Connect()
+        {
+            run = true;
+        }
+
+        public void Send(string message)
+        {
+
+            byte[] buffer = encoder.GetBytes(message);
+            if (buffer.Length > MAXPAYLOADSIZE)
```

Comment:
hmm were you running into issues with messages that were too large? I don't think you need to check max datagram size, since I'm assuming the C# UDP implementation handles the RFC specs and will split up the message into separate datagrams